sys/usb/usbus/cdc/acm/cdc_acm.c

```diff
@@ -126,6 +126,23 @@ static size_t _gen_full_acm_descriptor(usbus_t *usbus, void *arg)
 /* Submit (ACM interface in) */
 size_t usbus_cdc_acm_submit(usbus_cdcacm_device_t *cdcacm, const uint8_t *buf, size_t len)
 {
+    if (cdcacm->state == USBUS_CDC_ACM_LINE_STATE_DISCONNECTED) {
+        // stuff as much data as possible into tsrb, discarding the oldest
+        size_t n = tsrb_free(&cdcacm->tsrb);
+        if (len > n) {
+            tsrb_drop(
+                &cdcacm->tsrb,
+                len > cdcacm->tsrb.size ? cdcacm->tsrb.size : len - n
```

Comment:
You don't need this check. `tsrb_drop()` will exit once the `tsrb_empty()` condition is fulfilled.